diff -ruN tolua++-1.0.2/Makefile /home/tassi/Projects/liberopops/freepops/modules/src/tolua++-1.0.2/Makefile
--- tolua++-1.0.2/Makefile	2003-10-23 10:56:24.000000000 +0200
+++ /home/tassi/Projects/liberopops/freepops/modules/src/tolua++-1.0.2/Makefile	2004-01-22 17:47:46.000000000 +0100
@@ -1,5 +1,52 @@
-# makefile for tolua hierarchy
+include $(CONFIG)
+
+ifneq "$(OS)" "Darwin"
+	CPFLAGS = -a
+	LDFLAGS+= -Wl,-E
+	HLDFLAGS+= -Wl,-E
+endif
+
+CFLAGS += -I$(PREFIX)/include -I../../include
+HCFLAGS += -I$(PREFIX)/include -I../../include
+LDFLAGS += -Llib -L$(PREFIX)/lib -ltolua++-host -llua-host -llualib-host -ldl -lm
+HLDFLAGS += -Llib -L$(PREFIX)/lib -ltolua++-host -llua-host -llualib-host -ldl -lm
+
+SRC1=tolua.c toluabind.c 
+SRC2=tolua_event.c tolua_is.c tolua_map.c tolua_push.c tolua_to.c
+
+all: bin/tolua++
+
+bin/tolua++:
+	@echo -n " building library..."
+	@cd src/lib/;\
+		$(CC) $(CFLAGS) -c $(SRC2)
+	@$(AR) r lib/libtolua++.a src/lib/*.o
+	@$(RANLIB) lib/libtolua++.a
+	@echo "done."
+	
+	@if test "$(HCC)" = "$(CC)" ; then 		\
+		cd lib;					\
+		ln -s libtolua++.a libtolua++-host.a;	\
+	else						\
+		cd src/lib/;				\
+		echo -n " building host library...";	\
+		$(HCC) $(HCFLAGS) -c $(SRC2);		\
+		cd ../..;				\
+		$(HAR) r  lib/libtolua++-host.a src/lib/*.o;	\
+		$(HRANLIB) lib/libtolua++-host.a;		\
+		echo "done.";					\
+	fi
+
+	@echo -n " building tolua..."
+	@cd src/bin/;\
+		$(HCC) $(HCFLAGS) -c $(SRC1)
+	@$(HCC) -o bin/tolua++ src/bin/*.o $(HLDFLAGS)
+	@echo "done."
+
+	@cp include/*.h $(PREFIX)/include
+	@cp bin/* $(PREFIX)/bin
+	@cp $(CPFLAGS) lib/*.a $(PREFIX)/lib
+
+clean:
+	@rm -f bin/* src/*/*.o lib/*
 
-all:
-	@echo "Makefile is deprecated ;)"
-	@echo "see INSTALL for details on how to build tolua++"
