diff -ruN luasocket-2.0.2-old/makefile luasocket-2.0.2/makefile
--- luasocket-2.0.2-old/makefile	2007-10-14 01:44:03.000000000 +0200
+++ luasocket-2.0.2/makefile	1970-01-01 01:00:00.000000000 +0100
@@ -1,51 +0,0 @@
-#------
-# Load configuration
-#
-include config
-
-#------
-# Hopefully no need to change anything below this line
-#
-INSTALL_SOCKET_SHARE=$(INSTALL_TOP_SHARE)/socket
-INSTALL_SOCKET_LIB=$(INSTALL_TOP_LIB)/socket
-INSTALL_MIME_SHARE=$(INSTALL_TOP_SHARE)/mime
-INSTALL_MIME_LIB=$(INSTALL_TOP_LIB)/mime
-
-all clean:
-	cd src; $(MAKE) $@
-
-#------
-# Files to install
-#
-TO_SOCKET_SHARE:= \
-	http.lua \
-	url.lua \
-	tp.lua \
-	ftp.lua \
-	smtp.lua
-
-TO_TOP_SHARE:= \
-	ltn12.lua \
-	socket.lua \
-	mime.lua
-
-TO_MIME_SHARE:= 
-
-#------
-# Install LuaSocket according to recommendation
-#
-install: all
-	cd src; mkdir -p $(INSTALL_TOP_SHARE)
-	cd src; $(INSTALL_DATA) $(TO_TOP_SHARE) $(INSTALL_TOP_SHARE)
-	cd src; mkdir -p $(INSTALL_SOCKET_SHARE)
-	cd src; $(INSTALL_DATA) $(TO_SOCKET_SHARE) $(INSTALL_SOCKET_SHARE)
-	cd src; mkdir -p $(INSTALL_SOCKET_LIB)
-	cd src; $(INSTALL_EXEC) $(SOCKET_SO) $(INSTALL_SOCKET_LIB)/core.$(EXT)
-	#cd src; mkdir -p $(INSTALL_MIME_SHARE)
-	#cd src; $(INSTALL_DATA) $(TO_MIME_SHARE) $(INSTALL_MIME_SHARE)
-	cd src; mkdir -p $(INSTALL_MIME_LIB)
-	cd src; $(INSTALL_EXEC) $(MIME_SO) $(INSTALL_MIME_LIB)/core.$(EXT)
-
-#------
-# End of makefile
-#
diff -ruN luasocket-2.0.2-old/Makefile luasocket-2.0.2/Makefile
--- luasocket-2.0.2-old/Makefile	1970-01-01 01:00:00.000000000 +0100
+++ luasocket-2.0.2/Makefile	2008-06-06 19:24:35.000000000 +0200
@@ -0,0 +1,33 @@
+include $(CONFIG)
+
+H=@
+
+C= auxiliar.c buffer.c except.c inet.c io.c \
+	luasocket.c options.c select.c tcp.c \
+	timeout.c udp.c
+
+U_C = usocket.c
+W_C = wsocket.c
+
+all: src/libluasocket.a
+	$(H)echo -n
+
+src/libluasocket.a:
+	$(H)echo -n ' building luasocket... '
+	$(H)cd src ; 					\
+		$(CC) $(CFLAGS) -I $(PREFIX)/include $(MACHOARCH) -c $(C) ; 		\
+		if [ $(OS) = "Windows" ]; then \
+			$(CC) $(CFLAGS) -I $(PREFIX)/include $(MACHOARCH) -c $(W_C) ; 		\
+		else \
+			$(CC) $(CFLAGS) -I $(PREFIX)/include $(MACHOARCH) -c $(U_C) ; 		\
+		fi; \
+		$(AR) rc libluasocket.a *.o; 		\
+		$(RANLIB) libluasocket.a ;			\
+		rm -f *.o
+	$(H)echo "done."
+	$(H)cp src/libluasocket.a $(PREFIX)/lib
+	$(H)cp src/socket.lua $(PREFIX)/include
+	$(H)echo 'int luaopen_socket_core(lua_State*);'> $(PREFIX)/include/luasocket.h
+clean:
+	$(H)rm -f src/*.a src/*.o
+
