diff -ruN luaexpat-1.0.1/Makefile luaexpat-1.0.1-new/Makefile
--- luaexpat-1.0.1/Makefile	1970-01-01 01:00:00.000000000 +0100
+++ luaexpat-1.0.1-new/Makefile	2006-01-13 21:36:15.000000000 +0100
@@ -0,0 +1,35 @@
+include $(CONFIG)
+
+WARNS = -Wall -pedantic \
+        -Waggregate-return \
+        -Wcast-align \
+        -Wmissing-prototypes \
+        -Wstrict-prototypes \
+        -Wnested-externs \
+        -Wpointer-arith \
+        -Wshadow \
+        -Wwrite-strings
+
+
+CFLAGS += $(WARNS) -I$(PREFIX)/include -I. 
+
+LDFLAGS+= -L../lib/ -L../../lib/ -lexpat
+
+H=@
+
+all: libluaexpat.a
+	$(H)echo -n
+
+libluaexpat.a : src/lxplib.c
+	$(H)echo " compiling lxplib.c -> lxplib.o"
+	$(H)$(CC) $(CFLAGS) -c  src/lxplib.c -o src/lxplib.o -I src/
+	$(H)echo " linking libluaexpat.a"
+	$(H)$(AR) rc libluaexpat.a src/lxplib.o 1>/dev/null		
+	$(H)cp src/lxplib.h $(PREFIX)/include
+	$(H)mkdir -p $(PREFIX)/include/lxp/
+	$(H)cp src/lxp/lom.lua $(PREFIX)/include/lxp/
+	$(H)cp $(CPFLAGS) libluaexpat.a $(PREFIX)/lib
+
+clean:
+	$(H)rm -f *.a *.o
+.PHONY: all
diff -ruN luaexpat-1.0.1/src/lxp/lom.lua luaexpat-1.0.1-new/src/lxp/lom.lua
--- luaexpat-1.0.1/src/lxp/lom.lua	2005-06-03 22:18:23.000000000 +0200
+++ luaexpat-1.0.1-new/src/lxp/lom.lua	2006-01-13 22:13:23.000000000 +0100
@@ -7,7 +7,7 @@
 local assert, type, print = assert, type, print
 local lxp = lxp
 
-module ("lom")
+module ("lxp.lom")
 
 local function starttag (p, tag, attr)
   local stack = p:getcallbacks().stack
