diff -ruN luaexpat-1.0b/makefile luaexpat-1.0b-neq/makefile
--- luaexpat-1.0b/makefile	2004-04-04 19:53:33.000000000 +0200
+++ luaexpat-1.0b-neq/makefile	1970-01-01 01:00:00.000000000 +0100
@@ -1,41 +0,0 @@
-# Compilation parameters
-CC = gcc
-CWARNS = -Wall -pedantic \
-        -Waggregate-return \
-        -Wcast-align \
-        -Wmissing-prototypes \
-        -Wstrict-prototypes \
-        -Wnested-externs \
-        -Wpointer-arith \
-        -Wshadow \
-        -Wwrite-strings
-
-
-CFLAGS = $(CONFIG) $(CWARNS) -ansi -g -O2 -I/usr/local/include/lua5 \
-   -L./expat/xmlparse
-
-VERSION= 1.0b
-PKG = luaexpat-$(VERSION)
-TAR_FILE= $(PKG).tar.gz
-ZIP_FILE= $(PKG).zip
-SRCS= README makefile \
-	lxplib.c lxplib.h lxp.lua lom.lua \
-	test.lua test-lom.lua \
-	index.html manual.html lom.html luaexpat.png
-
-
-liblxp.so : lxplib.o
-	ld -o liblxp.so -shared lxplib.o -lexpat
-
-liblxp.dylib : lxplib.o
-	gcc -o liblxp.dylib -dynamiclib lxplib.o -lexpat -llua.5.0 -llualib.5.0
-
-clean:
-	rm -f liblxp.so liblxp.dylib lxplib.o
-
-dist:
-	mkdir $(PKG)
-	cp $(SRCS) $(PKG)
-	tar -czf $(TAR_FILE) $(PKG)
-	zip -lq $(ZIP_FILE) $(PKG)/*
-	rm -rf $(PKG)
--- luaexpat-1.0b/Makefile	1970-01-01 01:00:00.000000000 +0100
+++ luaexpat-1.0b-neq/Makefile	2004-08-14 20:41:04.000000000 +0200
@@ -0,0 +1,33 @@
+include $(CONFIG)
+
+WARNS = -Wall -pedantic \
+        -Waggregate-return \
+        -Wcast-align \
+        -Wmissing-prototypes \
+        -Wstrict-prototypes \
+        -Wnested-externs \
+        -Wpointer-arith \
+        -Wshadow \
+        -Wwrite-strings
+
+
+CFLAGS += $(WARNS) -I$(PREFIX)/include -I. 
+
+LDFLAGS+= -L../lib/ -L../../lib/ -lexpat
+
+H=@
+
+all: libluaexpat.a
+	$(H)echo -n
+
+libluaexpat.a : lxplib.c
+	$(H)echo " compiling lxplib.c -> lxplib.o"
+	$(H)$(CC) $(CFLAGS) -c  lxplib.c 
+	$(H)echo " linking libluaexpat.a"
+	$(H)$(AR) rc libluaexpat.a lxplib.o 1>/dev/null		
+	$(H)cp lxplib.h $(PREFIX)/include
+	$(H)cp $(CPFLAGS) libluaexpat.a $(PREFIX)/lib
+
+clean:
+	$(H)rm -f *.a *.o
+.PHONY: all
--- luaexpat-1.0b/lxplib.c	2004-04-04 19:53:33.000000000 +0200
+++ luaexpat-1.0b-neq/lxplib.c	2004-08-14 20:48:11.000000000 +0200
@@ -9,7 +9,7 @@
 #include <stdlib.h>
 #include <string.h>
 
-#include <xmlparse.h>
+#include <expat.h>
 
 #include <lua.h>
 #include <lauxlib.h>
@@ -30,7 +29,7 @@
 
 struct lxp_userdata {
   lua_State *L;
-  XML_Parser *parser;  /* associated expat parser */
+  XML_Parser parser;  /* associated expat parser */
   int tableref;  /* table with callbacks for this parser */
   enum XPState state;
   luaL_Buffer *b;  /* to concatenate sequences of cdata pieces */
@@ -41,7 +40,7 @@
 
 static int reporterror (lxp_userdata *xpu) {
   lua_State *L = xpu->L;
-  XML_Parser *p = xpu->parser;
+  XML_Parser p = xpu->parser;
   lua_pushnil(L);
   lua_pushstring(L, XML_ErrorString(XML_GetErrorCode(p)));
   lua_pushnumber(L, XML_GetCurrentLineNumber(p));
@@ -382,10 +381,10 @@
 }
 
 
-static lxp_userdata *checkparser (lua_State *L, int index) {
-  lxp_userdata *xpu = (lxp_userdata *)luaL_checkudata(L, index, ParserType);
-  luaL_argcheck(L, xpu, index, "expat parser expected");
-  luaL_argcheck(L, xpu->parser, index, "parser is closed");
+static lxp_userdata *checkparser (lua_State *L, int inde) {
+  lxp_userdata *xpu = (lxp_userdata *)luaL_checkudata(L, inde, ParserType);
+  luaL_argcheck(L, xpu, inde, "expat parser expected");
+  luaL_argcheck(L, xpu->parser, inde, "parser is closed");
   return xpu;
 }
 
@@ -474,7 +473,7 @@
 
 static int lxp_pos (lua_State *L) {
   lxp_userdata *xpu = checkparser(L, 1);
-  XML_Parser *p = xpu->parser;
+  XML_Parser p = xpu->parser;
   lua_pushnumber(L, XML_GetCurrentLineNumber(p));
   lua_pushnumber(L, XML_GetCurrentColumnNumber(p) + 1);
   lua_pushnumber(L, XML_GetCurrentByteIndex(p) + 1);
